ghcide/test/exe/Main.hs

```diff
@@ -6986,18 +6990,20 @@ positionMappingTests =
                     \(range, replacement, oldPos, newPos) ->
                     fromCurrent range replacement newPos === PositionExact oldPos
               , testProperty "toCurrent r t <=< fromCurrent r t" $ do
-                let gen = do
+                let gen :: Gen (Rope, Range, T.Text)
+                    gen = do
                         rope <- genRope
                         range <- genRange rope
                         PrintableText replacement <- arbitrary
-                        let newRope = applyChange rope (TextDocumentContentChangeEvent (Just range) Nothing replacement)
-                        newPos <- genPosition newRope
-                        pure (range, replacement, newPos)
-                forAll
-                    (suchThatMap gen
-                        (\(range, replacement, newPos) -> positionResultToMaybe $ (range, replacement, newPos,) <$> fromCurrent range replacement newPos)) $
-                    \(range, replacement, newPos, oldPos) ->
-                    toCurrent range replacement oldPos === PositionExact newPos
+                        pure (rope, range, replacement)
+                monadicIO $ forAllM gen
+                    $ \(rope, range, replacement) -> do
+                        newRope <- liftIO $ applyChange (toCologActionWithPrio $ cmapWithPrio LogVfs recorder) rope (TextDocumentContentChangeEvent (Just range) Nothing replacement)
```

Comment:
Why not pass `mempty :: LogAction Identity` here? then you don't need IO, and you'll just not get hte log messages which you don't care about anyway. Then I think you can keep the old code.

Reply:
Good advice!